File: 2021/day2/day2.go
package day2

import (
	"fmt"
	"log"
	"strconv"
	"strings"
)

func Run(input string) {
	fmt.Println("Day 2 Part 2:", Part1(input))
	fmt.Println("Day 2 Part 2:", Part2(input))
}

func Part2(input string) int {
	lines := strings.Split(input, "\n")

	depth := 0
	horizontal := 0
	aim := 0
	for _, command := range lines {
		splittedCommand := strings.Split(command, " ")

		position, err := strconv.Atoi(splittedCommand[1])
		if err != nil {
			log.Fatal(err)
		}

		switch splittedCommand[0] {
		case "forward":
			horizontal += position
			depth += (position * aim)
		case "down":
			aim += position
		case "up":
			aim -= position
		}
	}

	return depth * horizontal
}

func Part1(input string) int {
	lines := strings.Split(input, "\n")

	depth := 0
	horizontal := 0
	for _, command := range lines {
		splittedCommand := strings.Split(command, " ")

		position, err := strconv.Atoi(splittedCommand[1])
		if err != nil {
			log.Fatal(err)
		}

		switch splittedCommand[0] {
		case "forward":
			horizontal += position
		case "down":
			depth += position
		case "up":
			depth -= position
		}
	}

	return depth * horizontal
}

File: 2021/day2/day2.py
# day2.py


def run(input: str):
    print(f"Day 2 Part 1: {part1(input)}")
    print(f"Day 2 Part 2: {part2(input)}")


def part2(input: str):
    lines = input.splitlines()

    depth = 0
    horizontal = 0
    aim = 0

    for line in lines:
        splitted_command = line.split(" ")

        command = splitted_command[0]
        position = int(splitted_command[1])

        if command == "up":
            aim -= position
        elif command == "down":
            aim += position
        elif command == "forward":
            horizontal += position
            depth += position * aim

    return depth * horizontal


def part1(input: str):
    lines = input.splitlines()

    depth = 0
    horizontal = 0

    for line in lines:
        splitted_command = line.split(" ")

        command = splitted_command[0]
        position = int(splitted_command[1])

        if command == "up":
            depth -= position
        elif command == "down":
            depth += position
        elif command == "forward":
            horizontal += position

    return depth * horizontal
